codespell_lib/_codespell.py

```diff
@@ -35,8 +35,16 @@
 
 # Users might want to link this file into /usr/local/bin, so we resolve the
 # symbolic link path to the real path if necessary.
-default_dictionary = os.path.join(os.path.dirname(os.path.realpath(__file__)),
-                                  'data', 'dictionary.txt')
+_data_root = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+_builtin_dictionaries = (  # name, desc, name, should be in aspell
```

Comment:
If we track the dictionary a typo came from, we can also add another column here which is either whether to offer itself as a suggestion, or whether to allow a replacement of itself. Which should probably be true for all but clear if you're in interactive mode.

Adding the error itself as a typo is probably easiest, but then we want to make the error the correct capitalisation and lower case it, e.g. `Tim->time` becomes `tim->time, Tim` in interactive mode, but the downside is that stops us from suggesting case changes in future (to correct e.g. WI-Fi to Wi-Fi), but we have to do a bit more dictionary work if not, but we can chose the order of the suggestions including it.